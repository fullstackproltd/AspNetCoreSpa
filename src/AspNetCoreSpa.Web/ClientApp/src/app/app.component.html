<!-- app modal component -->
<ng-template modal>
  <appc-modal-component></appc-modal-component>
</ng-template>

<!-- header component -->
<appc-header></appc-header>

<!-- component routing placeholder -->
<main role="main">
  <router-outlet #o="outlet"></router-outlet>
</main>

<!-- footer component -->
<appc-footer></appc-footer>
