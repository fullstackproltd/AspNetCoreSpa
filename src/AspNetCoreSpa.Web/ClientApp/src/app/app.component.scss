main {
    margin-top: 60px;
}