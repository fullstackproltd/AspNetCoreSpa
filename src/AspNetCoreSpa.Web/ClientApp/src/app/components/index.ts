export * from './footer';
export * from './header';
export * from './modal';
export * from './privacy';
