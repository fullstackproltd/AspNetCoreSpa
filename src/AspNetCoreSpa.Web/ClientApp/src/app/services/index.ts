export * from './interceptors';
export * from './app.service';
export * from './data.service';
export * from './auth.service';
export * from './modal.service';
export * from './modal-state.service';
export * from './global-error.service';
