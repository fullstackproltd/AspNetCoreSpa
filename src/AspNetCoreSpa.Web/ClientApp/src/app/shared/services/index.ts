export * from './custom-date-formatter.service';
export * from './custom-date-adapter.service';
