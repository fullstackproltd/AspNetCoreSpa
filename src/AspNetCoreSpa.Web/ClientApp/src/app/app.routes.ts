import { Routes } from '@angular/router';
import { HomeComponent } from '@app/home/home.component';
import { PrivacyComponent } from '@app/components';

export const routes: Routes = [
  { path: '', component: HomeComponent, pathMatch: 'full', data: { state: 'home' } },
  { path: 'examples', loadChildren: './+examples/examples.module#ExamplesModule' },
  { path: 'privacy', component: PrivacyComponent }
];
