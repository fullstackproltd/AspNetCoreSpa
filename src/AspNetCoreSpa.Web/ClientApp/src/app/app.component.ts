import { Component, OnInit } from '@angular/core';
import { Title, Meta } from '@angular/platform-browser';
import { Router } from '@angular/router';

// import { routerTransition } from './router.animations';
import { AppService, AuthService, ModalService } from '@app/services';
@Component({
  selector: 'appc-root',
  // animations: [routerTransition],
  styleUrls: ['./app.component.scss'],
  templateUrl: './app.component.html'
})
export class AppComponent implements OnInit {
  constructor(
    private authService: AuthService,
    private router: Router,
    private title: Title,
    private meta: Meta,
    private appService: AppService,
    private modalService: ModalService
  ) { }

  public ngOnInit() {
    this.updateTitleAndMeta();
    if (window.location.href.indexOf('?postLogout=true') > 0) {
      this.authService.signoutRedirectCallback().then(() => {
        const url: string = this.router.url.substring(
          0,
          this.router.url.indexOf('?')
        );
        this.router.navigateByUrl(url);
      });
    }
    // Check cookie consent
    setTimeout(() => {
      if (this.appService.appData.cookieConsent.showConsent) {
        this.modalService.confirm({
          title: 'Cookie consent',
          message: 'Use this space to summarize your privacy and cookie use policy.'
        }).then(() => {
          document.cookie = this.appService.appData.cookieConsent.cookieString;
        }, () => { });
      }
    }, 0);

  }

  public getState(outlet: any) {
    return outlet.activatedRouteData.state;
  }

  private updateTitleAndMeta() {
    this.title.setTitle(this.appService.appData.content['app_title']);
    this.meta.addTags([
      { name: 'description', content: this.appService.appData.content['app_description'] },
      { property: 'og:title', content: this.appService.appData.content['app_title'] },
      { property: 'og:description', content: this.appService.appData.content['app_description'] }
    ]);
  }
}
