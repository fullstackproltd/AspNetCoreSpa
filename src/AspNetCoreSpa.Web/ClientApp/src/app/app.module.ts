import { BrowserModule, BrowserTransferStateModule } from '@angular/platform-browser';
import { NgModule, APP_INITIALIZER, ErrorHandler } from '@angular/core';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { RouterModule } from '@angular/router';
import { HTTP_INTERCEPTORS, HttpClientModule } from '@angular/common/http';
import { ServiceWorkerModule } from '@angular/service-worker';
import { NgbModule } from '@ng-bootstrap/ng-bootstrap';

// import { PrebootModule } from 'preboot';

import { environment } from '../environments/environment';

import { AppSharedModule } from './appshared';
import { ToastrModule } from './toastr';

import { routes } from './app.routes';
// Components
import { FooterComponent, HeaderComponent, ModalComponent, PrivacyComponent, ModalTemplateDirective } from '@app/components';
import { AppComponent } from './app.component';
import { HomeComponent } from './home/home.component';
// Services
import { AppService, AuthService, DataService, GlobalErrorHandler, ModalService, ModalStateService, AuthInterceptor, TimingInterceptor } from '@app/services';
export function appServiceFactory(appService: AppService, authService: AuthService): Function {
  return () => appService.getAppData(authService);
}
@NgModule({
  declarations: [
    // Components
    AppComponent,
    HomeComponent,
    FooterComponent,
    HeaderComponent,
    ModalComponent,
    ModalTemplateDirective,
    PrivacyComponent
  ],
  imports: [
    BrowserModule.withServerTransition({ appId: 'ng-cli-universal' }),
    // PrebootModule.withConfig({ appRoot: 'appc-root' }),
    BrowserAnimationsModule,
    BrowserTransferStateModule,
    HttpClientModule,
    AppSharedModule,
    // OAuthModule.forRoot(),
    NgbModule.forRoot(),
    ToastrModule.forRoot(),
    RouterModule.forRoot(routes, { initialNavigation: 'enabled' }),
    ServiceWorkerModule.register('ngsw-worker.js', { enabled: environment.production }),
  ],
  providers: [
    AppService,
    AuthService,
    DataService,
    ModalService,
    ModalStateService,
    { provide: HTTP_INTERCEPTORS, useClass: AuthInterceptor, multi: true },
    { provide: HTTP_INTERCEPTORS, useClass: TimingInterceptor, multi: true },
    { provide: APP_INITIALIZER, useFactory: appServiceFactory, deps: [AppService, AuthService], multi: true },
    { provide: ErrorHandler, useClass: GlobalErrorHandler }
  ],
  exports: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
