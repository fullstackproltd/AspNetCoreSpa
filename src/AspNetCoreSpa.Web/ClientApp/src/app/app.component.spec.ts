import { NoopAnimationsModule } from '@angular/platform-browser/animations';
import { TestBed, async } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from './app.component';
import { AppService } from '@app/services';
import { MockAppService } from '../../__mocks__/MockAppService';

describe('AppComponent', () => {
  let fixture;

  beforeEach(
    async(() => {
      TestBed.configureTestingModule({
        declarations: [AppComponent],
        imports: [
          NoopAnimationsModule,
          RouterTestingModule.withRoutes([])
        ],
        schemas: [NO_ERRORS_SCHEMA],
        providers: [
          { provide: AppService, useClass: MockAppService },
          { provide: 'BASE_URL', useValue: 'http://mock.com' },
        ]
      }).compileComponents();
    })
  );

  beforeEach(() => {
    fixture = TestBed.createComponent(AppComponent);
    fixture.detectChanges();
  });

  it('should be created without error',
    async(() => {
      const app = fixture.debugElement.componentInstance;
      expect(app).toBeTruthy();
    })
  );

  it('snapshot created successfully', () => {
    expect(fixture).toMatchSnapshot();
  });

});
