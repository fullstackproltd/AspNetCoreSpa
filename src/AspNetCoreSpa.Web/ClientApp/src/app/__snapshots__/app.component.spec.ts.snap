// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppComponent snapshot created successfully 1`] = `
<appc-root
  appService={[Function MockAppService]}
  authService={[Function MockAuthService]}
  meta={[Function Meta]}
  modalService={[Function ModalService]}
  router={[Function Router]}
  title={[Function Title]}
>
  <appc-header /><main
    role="main"
  >
    <router-outlet />
  </main><appc-footer />
</appc-root>
`;
